api/v1alpha1/virtualmachineclass_types.go

```diff
@@ -8,10 +8,29 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// VGPUDevice contains the configuration corresponding to a vGPU device.
+type VGPUDevice struct {
+	ProfileName string `json:"profileName,omitempty"`
+}
+
+// DynamicDirectPathIODevice contains the configuration corresponding to a Dynamic DirectPath I/O device.
+type DynamicDirectPathIODevice struct {
+	VendorId    int    `json:"vendorId,omitempty"`
```

Comment:
`ID` instead of `Id` matches k8s convention

Does `omitempty` really apply to all of these?